platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -26,7 +28,24 @@ class MGLStyleValueTransformer {
             return nil;
         }
     }
-    
+
+    MGLStyleValue<ObjCType> *toLineJoinStyleValue(const mbgl::style::PropertyValue<MBGLType> &mbglValue) {
+        auto str = mbgl::Enum<MBGLType>::toString(mbglValue.asConstant());
+        NSString *enumString = [NSString stringWithCString:str encoding:NSUTF8StringEncoding];
+        NSUInteger enumValue;
+        if ([enumString isEqualToString:@"miter"]) {
```

Comment:
Nice! Using `mbgl::Enum` is a lot cleaner than what I thought we’d have to resort to (the `_names` macros).

I think we could take further advantage of `mbgl::Enum`, using `MBGL_DEFINE_ENUM()` to map e.g. `{ MGLLineJoinMiter, "miter" }` and calling `toEnum()` on the string. That way, we can avoid going the NSString conversion and these if statements, and we can generate everything using the style specification (in a separate file).